frontend/src/components/PublishTags.tsx

```diff
@@ -5,134 +5,160 @@ import { BACKEND_URL } from "../config";
+      navigate(`/blog/${blogId}`);
+    } catch (e) {
+      return { error: "An error has occured trying to edit the blog" };
+    }
+  }
 
-	return (
-		<>
-			<button
-				className="primary"
-				type="button"
-				disabled={loading}
-				onClick={async () => {
-					setLoading(true);
-					await onClick();
-					setVisibility(true);
-					setLoading(false);
-				}}
-			>
-				<div className="flex items-center gap-2">
-					{loading && <Spinner className="h-4 w-4 !border-2" />}
-					Publish
-				</div>
-			</button>
-			{visible && (
-				<div
-					aria-hidden="true"
-					className={`${
-						visible ? "block" : "hidden"
-					} overflow-y-auto overflow-x-hidden fixed top-0 right-0 left-0 z-50 justify-center items-center w-full md:inset-0 h-[calc(100%-1rem)] max-h-full`}
-				>
-					<div className="relative w-full h-[100vh] max-h-full">
-						<div className="relative h-full bg-white rounded-lg shadow dark:bg-gray-700">
-							<div className="flex items-center justify-between p-4 md:p-5 border-b rounded-t dark:border-gray-600">
-								<button
-									type="button"
-									className="text-gray-400 bg-transparent hover:bg-gray-200 hover:text-gray-900 rounded-lg text-sm w-8 h-8 ms-auto inline-flex justify-center items-center dark:hover:bg-gray-600 dark:hover:text-white"
-									data-modal-hide="default-modal"
-									onClick={() => setVisibility(false)}
-								>
-									<svg
-										className="w-3 h-3"
-										aria-hidden="true"
-										xmlns="http://www.w3.org/2000/svg"
-										fill="none"
-										viewBox="0 0 14 14"
-									>
-										<path
-											stroke="currentColor"
-											stroke-linecap="round"
-											stroke-linejoin="round"
-											stroke-width="2"
-											d="m1 1 6 6m0 0 6 6M7 7l6-6M7 7l-6 6"
-										/>
-									</svg>
-									<span className="sr-only">Close modal</span>
-								</button>
-							</div>
-							<div className="p-4 md:p-5 space-y-4 flex flex-col justify-center items-center">
-								<div className="md:w-[500px]">
-									<div className="py-4 text-lg">
-										Publishing to: <span className="font-bold">Aadesh Kulkarni</span>
-									</div>
-									<p>Add or change topics (up to 5) so readers know what your story is about</p>
-									<MultiSelectDropdown
-										options={tagOptions}
-										selectedOptions={selectedOptions}
-										setSelectedOptions={setSelectedOptions}
-									/>
-									<div className="flex gap-4 items-center p-4 md:p-5 border-t border-gray-200 rounded-b dark:border-gray-600">
-										<button type="button" className="primary" onClick={linkTopics}>
-											Publish
-										</button>
-										<button type="button" className="secondary">
-											Cancel
-										</button>
-									</div>
-								</div>
-							</div>
-						</div>
-					</div>
-				</div>
-			)}
-		</>
-	);
+  return (
+    <>
+      <button
+        className="primary"
+        type="button"
+        disabled={loading}
```

Comment:
We should disable the publish button by default.

Only when title's length AND content's length becomes > 0, we can enable it.

Reply:
yes that was my initial solution but I went ahead with just giving a toast message since a similar flow is happening in edit blog screen. 
If we are disabling the publish button, we should also disable edit button when they remove the title and content. that way the UX will be consistent.
Let me know your thoughts.

Reply:
Alright.
Yes, UX consistency is as important as UX. 

We will need to change it for edit as well. Can you accommodate it in this PR or should I create a new issue ?

Reply:
I can make the changes for edit screen in this PR.
Also, if we are disabling the button altogether, do we need a check for opening the topics popup?

Reply:
Wouldn't be needed, no ? 
If the button is responsible to trigger the topic and the button itself is disabled, user wouldn't be able to access the popup until button is enabled.